### DISCUSSION TITLE
How to start this assignment

### DISCUSSION THREAD

**Discussion by Sapnajha19** [open-sauced-craftwork/browser-extension#34] [Q&A] [answered]
I don't understand how to start. Am I supposed to clone the browser extension repo first? [this](https://github.com/open-sauced-craftwork/open_sauced_craftwork_browser_extension-Sapnajha19) or do I have to start by running the commands in the [CRXJS](https://crxjs.dev/vite-plugin/getting-started/vanilla-js/create-project). Can you please explain step by step? And where is the classroom where we have to initialize the assignment?
@bdougie

**Comment by thekingslee** [comment]
I think you should clone the repo first. Then follow the CRXJS guide to set up the project

**Reply by Sapnajha19** [reply]
> I think you should clone the repo first. Then follow the CRXJS guide to set up the project

Am I supposed to select my preferred framework while setting up the project with CRXJS guide or should I use vanilla js? Is there any preference for a framework for this project that we should use?

**Reply by thekingslee** [reply]
A preferred framework was not mentioned on the assignment. I think you can go with vanilla JS if it's what you're most confident with.

**Reply by bdougie** [reply]
some folks have used frameworks, and that is ok. This assignment is intentionally vague on that to see your preference.

**Comment by anush008** [comment, marked as answer]
## Here's how to go about the assignment:
• You'll have to create a new branch, say "development", in the private repository created for you.
• Clone the newly created branch locally.
• Start working on the assignment.
• Push the code to the "development" branch you had created and cloned from earlier.
• Once you are done, create a PR to merge your "development" branch to the main branch and request a review from @bdougie in the PR.

He'll then be reviewing your submission and suggest changes if any.

**Reply by Sapnajha19** [reply]
> ## Here's how to go about the assignment:
> • You'll have to create a new branch, say "development", in the private repository created for you. • Clone the newly created branch locally. • Start working on the assignment. • Push the code to the "development" branch you had created and cloned from earlier. • Once you are done, create a PR to merge your "development" branch to the main branch and request a review from @bdougie in the PR.
> 
> He'll then be reviewing your submission and suggest changes if any.

Thank you , it helped

**Comment by Sapnajha19** [comment]
Do I have to set up and install packages in the main branch or the newly created branch?

**Reply by anush008** [reply]
All changes should be made in the branch that you created, while the main branch should only have a pull request for the purpose of review.
